### DISCUSSION TITLE
Benefit of using "compressed images"

### DISCUSSION THREAD

**Discussion by krangor** [buttondown/discussions#28] [Q&A] [answered]
Hi, I noticed the option to used "compressed" or "original" images in the image editor. As the images themselves are hosted in/on a CDN and not actually embedded in the email, is there a noticeable benefit here? As well, it's unclear to me what sort of quality loss this will entail. Left me a bit confused about the optimal choice. Thanks!

**Comment by jmduke** [comment]
@krangor Totally reasonable question. Even though we're CDN-ing the images and not embedding them in an email, really big images can still hurt readability/performance — both on the web archive side and the client side (GMail, for instance, dings emails that have multiple large images even if those images aren't render-blocking!)

**Reply by krangor** [reply]
Ah, interesting. Two questions then: how much of a hit does the image take, quality-wise, and is the original image still stored in the CDN but presented as a more bite-sized serving when the mail client calls for it?

**Reply by krangor** [reply]
(Also is there a floor, like if I upload an already-small image will it try and compress it even further or will it be cool with it as is?)

**Reply by jmduke** [reply]
1. We downscale to 960px; it's fairly naive. (Mostly useful for folks who are doing things like uploading completely uncompressed RAW/JPGs.)
2. We don't upscale, so small images will be no-oped!

**Reply by krangor** [reply]
Ah so it's more a matter of resizing than recompressing/transcoding (to borrow a video term I'm not sure applies here)?

**Reply by jmduke** [reply]
Precisely! To get into the weeds a bit, we're using Cloudflare's Image Workers on the backend which do do something more clever than _just_ resizing, but we're not aggressively reconverting or anything like that

**Comment by krangor** [comment]
Great to know. Had nightmares of mid-2000s Facebook image recompression.

**Reply by jmduke** [reply, marked as answer]
And to the point of getting back the 'original' URL: here's really all that switch does behind the scenes. You can simply remove the queryparams we add to go from 'optimized' to 'unoptimized'

![image](https://github.com/buttondown/discussions/assets/461046/2eb7f525-f4d7-419b-961e-a5f36ca1964b)
